add compat behaviour for RISCV for RISCVC/RISCV_C constant (#2996)
aquynh/capstone: docs/cs_v6_release_guide.md

1. `arm64.h` is a compatibility header now, which merely maps every member to the one in the `aarch64.h` header. Defining `CAPSTONE_AARCH64_COMPAT_HEADER` before including `capstone.h` will include the headers in the right order.
2. The `systemz.h` header includes the `systemz_compatibility.h` header if `CAPSTONE_SYSTEMZ_COMPAT_HEADER` is defined.
3. Defining `CAPSTONE_RISCV_COMPAT_HEADER` before including `capstone.h` exposes the legacy RISC-V compressed-mode constant `CS_MODE_RISCVC` as an alias of `CS_MODE_RISCV_C`.

> We will continue to maintain both compatibility headers, `arm64.h` and `systemz_compatibility.h`.

Reviewer: ```suggestion
We will continue to maintain compatibility for `arm64.h`, `systemz_compatibility.h`, and `riscv.h`.
```

Reviewer: Or simply "for all compatibility headers".

Author: RISC-V is a bit different, I didn't create a dedicated header like arm or systemz, the sole compatibility option is a single `#define`, I added it directly to the main header.

That's why I rephrased to indicate which headers will be maintained.

_Compatibility header_

> If you want to use the compatibility header and stick with the `ARM64`/`SYSZ` naming, you can define `CAPSTONE_AARCH64_COMPAT_HEADER` and `CAPSTONE_SYSTEMZ_COMPAT_HEADER` before including `capstone.h`. For the legacy RISC-V compressed-mode spelling, define `CAPSTONE_RISCV_COMPAT_HEADER` before including `capstone.h`.

Reviewer: ```suggestion
If you want to use the compatibility header and stick with the `ARM64`/`SYSZ` naming, you can define `CAPSTONE_AARCH64_COMPAT_HEADER` and `CAPSTONE_SYSTEMZ_COMPAT_HEADER` before including `capstone.h`. For the legacy RISC-V compressed-mode spelling (`CS_MODE_RISCVC`), define `CAPSTONE_RISCV_COMPAT_HEADER` before including `capstone.h`.
```
